database/generateData.js

```diff
@@ -1,55 +1,65 @@
 const fs = require('fs');
 const faker = require('faker');
 
-const write = function(filename, numRecords) {
-  return new Promise((resolve, reject) => {
+let name;
+let numRec;
+let i = 1;
+let stream;
+let callback;
+
+
+const write = function(filename, numRecords, writeStream, callback) {
+    name = filename || name;
+    numRec = numRecords || numRec;
+    stream = writeStream || stream;
+    callback = callback || function() {
+      console.log('done!');
+    };
```

Comment:
Currently the variables being used as default are undefined. Also, variable "callback" is masked by the argument to the function.

Reply:
Yeah i wanted to set them on the first run before the initial recursion to prevent re-initialization

Reply:
That makes sense. Personally I would use a anonymous function (which calls the write function with the same arguments) as the callback given to the drain event, so you can avoid closure variables. E.g. stream.on('drain', () => write(same, args, here))